plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+	}
+
+	/**
+	 * Returns a cached Press or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedPress($pressPath) {
+		return $this->_presses[$pressPath] ?? $this->_presses[$pressPath] = $this->getPressByPath($pressPath);
+	}
+
+	/**
+	 * Returns a cached Genre or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedGenreId($pressId) {
+		return $this->_genreIds[$pressId] ?? $this->_genreIds[$pressId] = $this->getGenreIdByPressId($pressId);
+	}
+
+	/**
+	 * Returns a cached Serie ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $seriesPath
+	 * @param int $pressId
+	 */
+	private function getCachedSerieId($seriesPath, $pressId) {
+		$key = "{$seriesPath}_{$pressId}";
+		return $this->_serieIds[$key] ?? $this->_serieIds[$key] = $this->getSerieIdByPathAndPressId($seriesPath, $pressId);
+	}
+
+	/**
+	 * Returns a cached UserGroup ID or create a new one, if it isn't retreived yet.
+	 *
+	 * @param string $pressPath
+	 */
+	private function getCachedUserGroupId($pressId) {
+		return $this->_userGroupIds[$pressId] ?? $this->_userGroupIds[$pressId] = $this->getUserGroupIdByPressId($pressId);
+	}
+
+	/**
+	 * Insert data on the invalid_rows.csv file and increase the failed rows counter
+	 *
+	 * @param SplFileObject $invalidRowsFile
+	 * @param array $fields
+	 * @param int $failedRows
+	 * @param string $reason
+	 */
+	private function processFailedRow($invalidRowsFile, $fields, $reason, $failedRows) {
+		$invalidRowsFile->fputcsv(array_merge(array_pad($fields, $this->_expectedRowSize, null), [$reason]));
+
+		return $failedRows + 1;
+	}
+
+	/**
+	 * Verify if all required fields are present on a CSV row
+	 *
+	 * @param object $row
+	 */
+	private function requiredFieldsPresent($row) {
+		 return !!$row->pressPath
+		 	&& !!$row->authorString
+			&& !!$row->title
+			&& !!$row->abstract
+			&& !!$row->locale
+			&& !!$row->filename;
+	}
+
+	/**
+	 * Retrives a user by username
+	 *
+	 * @param string $username
+	 */
+	private function getUser($username) {
+		/** @var UserDAO $userDao  */
+		$userDao = $this->getCachedDao('UserDAO');
+		return $userDao->getByUsername($username);
+	}
+
+	/**
+	 * Retrieves a Press by path.
+	 *
+	 * @param string $pressPath
+	 * @return Press
+	 */
+	private function getPressByPath($pressPath) {
+		/** @var PressDAO $pressDao */
+		$pressDao = $this->getCachedDao('PressDAO');
+		return $pressDao->getByPath($pressPath);
+	}
+
+	/**
+	 * Retrieves a Genre ID by Press ID. If the Genre doesn't exist, the result
+	 * will be false
+	 *
+	 * @param int $pressId
+	 */
+	private function getGenreIdByPressId($pressId) {
+		/** @var GenreDAO $genreDao */
+		$genreDao = $this->getCachedDao('GenreDAO');
+		$genre = $genreDao->getByKey('MANUSCRIPT', $pressId);
+
+		return !$genre ? false : $genre->getId();
```

Comment:
Hmm, I think the genre name should be also a field on the CSV file, and if not found, then a default one could be passed by argument (if nothing was passed, then we can use the "MANUSCRIPT" as default).

```suggestion
	 * Retrieves the "Manuscript" genre's ID of a given Press ID
	 *
	 * @param int $pressId
	 * @return ?int Null if not found
	 */
	private function getManuscriptGenreId($pressId) {
		/** @var GenreDAO $genreDao */
		$genreDao = $this->getCachedDao('GenreDAO');
		$genre = $genreDao->getByKey('MANUSCRIPT', $pressId);

		return $genre ? $genre->getId() : null;
```